pandas/tests/frame/test_indexing.py

```diff
@@ -1944,6 +1944,29 @@ def test_set_value_resize(self, float_frame):
         with pytest.raises(ValueError, match=msg):
             res3._set_value("foobar", "baz", "sam")
 
+    def test_loc_indexing_preserves_index_dtype(self):
```

Comment:
Can you mention categorical in the title?

Reply:
Thanks, @mroeschke - done.